lib/core/bloc/settings_cubit.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:bloc/bloc.dart';
+import 'package:kahootify/core/data/shared_preferences_repository.dart';
+import 'package:kahootify/core/models/settings.dart';
+
+class SettingsCubit extends Cubit<Settings> {
+  final SharedPreferencesRepository prefs;
+
+  SettingsCubit(this.prefs) : super(Settings.initial());
+
+  void initialize() {
+    final String? playerName = prefs.playerName;
+    final bool? enabledMusic = prefs.enabledMusic;
+    final bool? enabledSound = prefs.enabledSound;
+    emit(Settings.initial().copyWith(playerName: playerName, enabledSound: enabledSound, enabledMusic: enabledMusic));
+    if (enabledSound == null) {
+      prefs.enabledMusic = true;
+      prefs.enabledSound = true;
+    }
+  }
+
+  void changePlayerName(String playerName) {
```

Comment:
I would personally use set instead of change. Change might be associated with some database/server request. Set is indicating the variable.

Reply:
done